examples/mugen/retrieval/video_clip.py

```diff
@@ -76,36 +87,153 @@ def forward(self, input_ids: torch.Tensor) -> torch.Tensor:
 
 
 class VideoEncoder(nn.Module, PretrainedMixin):
-    """Encode videos to a fixed size vector.
+    """Encode videos to the last layer before the fully-connected layer of S3D.
 
     Adapted from MUGEN's video encoder
         (https://github.com/mugen-org/MUGEN_baseline/blob/main/lib/models/videoclip/modules.py)
 
     Args:
         pretrained (bool): whether to use a pretrained model or not.
             Defaults to True.
+        trainable (bool): whether the encoder's weights should be trainable (not frozen).
+            Defaults to True.
+            Warning: If ``pretrained`` is False, ``trainable`` will act as True regardless of the supplied value.
         pretrain_path (str): local path or remote URL to pretrained weights.
             Defaults to ``PRETRAINED_S3D_KINETICS400_URL``, the weights MUGEN used from
             pretraining S3D on Kinetics 400. Ignored if ``pretrained`` is False.
 
     Inputs:
         x (Tensor): batch of videos with dimensions (batch, channel, time, height, width)
+            Size of ``channel`` dimension must be 3.
+
     """
 
     def __init__(
         self,
         pretrained: bool = True,
+        trainable: bool = True,
         pretrain_path: str = PRETRAINED_S3D_KINETICS400_URL,
     ):
         super().__init__()
         self.model = S3D(400)
+        self.out_dim = self.model.fc.in_channels
         self.model.fc = nn.Identity()
 
         if pretrained:
             self.load_model(pretrain_path)
```

Comment:
nit: Add a prompt here for the user (the same for TextEncoder as for distilledBERT)

https://github.com/mugen-org/MUGEN_baseline/blob/main/lib/models/videoclip/modules.py#L80